File: dmn-test-cases/tck/1.1/cl3/0016-some-every/translator/expected/java/dmn/type/TItemPriceImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tItemPrice"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TItemPriceImpl implements TItemPrice {
        private String itemName;
        private java.math.BigDecimal price;

    public TItemPriceImpl() {
    }

    public TItemPriceImpl(String itemName, java.math.BigDecimal price) {
        this.setItemName(itemName);
        this.setPrice(price);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("itemName")
    public String getItemName() {
        return this.itemName;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("itemName")
    public void setItemName(String itemName) {
        this.itemName = itemName;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("price")
    public java.math.BigDecimal getPrice() {
        return this.price;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("price")
    public void setPrice(java.math.BigDecimal price) {
        this.price = price;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0016-some-every/type/TItemPriceImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tItemPrice"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TItemPriceImpl : TItemPrice {
    @get:com.fasterxml.jackson.annotation.JsonGetter("itemName")
    @set:com.fasterxml.jackson.annotation.JsonGetter("itemName")
    override var itemName: String? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("price")
    @set:com.fasterxml.jackson.annotation.JsonGetter("price")
    override var price: java.math.BigDecimal? = null

    constructor() {
    }

    constructor (itemName: String?, price: java.math.BigDecimal?) {
        this.itemName = itemName
        this.price = price
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
